plugins/include/tempent.inc

```diff
@@ -0,0 +1,1289 @@
+// vim: set ts=4 sw=4 tw=99 noet:
+//
+// AMX Mod X, based on AMX Mod by Aleksander Naszko ("OLO").
+// Copyright (C) The AMX Mod X Development Team.
+//
+// This software is licensed under the GNU General Public License, version 3 or higher.
+// Additional exceptions apply. For full license details, see LICENSE.txt or visit:
+//     https://alliedmods.net/amxmodx-license
+
+//
+// Temporary Entity Functions
+//
+
+#if defined _tempent_included
+	#endinput
+#endif
+
+#define _tempent_included
+
+/**
+ * Creates a beam between two points.
+ *
+ * @note A common sprite to use is "sprites/laserbeam.spr"
+ * @note Video preview of this and all other te_ natives can be found here:
+ *       https://youtu.be/szW-bSMPuyQ?t=3s
+ *
+ * @param startpos      Starting coordinates of the beam
+ * @param endpos        Ending coordinates of the beam
+ * @param sprite        The sprite index to use in the beam
+ * @param startframe    The frame to start with in the sprite (0 - 255)
+ * @param framerate     The frame rate to show the sprite at (0 - 255)
+ * @param life          The length of time the beam shall remain (0 - 255)
+ * @param width         The width of the beam (0 - 255)
+ * @param noise         The noise amplitude of the beam, this controls
+ *                      the distorsion of the beam (0 - 255)
+ * @param r             Red color amount (0 - 255)
+ * @param g             Green color amount (0 - 255)
+ * @param b             Blue color amount (0 - 255)
+ * @param a             Beam brightness (alpha) (0 - 255)
+ * @param speed         The scroll speed of the beam (0 - 255)
+ * @param receiver      Client index that will be able to see the beam
+ *                      or 0 for all clients
+ * @param reliable      If true, the message will be sent via the reliable
+ *                      channel, otherwise it will use the unreliable one
+ *
+ * @noreturn
+ * @error               If "receiver" is non-zero and the client isn't
+ *                      connected, an error will be thrown.
+ */
+native te_create_beam_between_points(startpos[3], endpos[3], sprite, startframe = 0, framerate = 30, life = 10, width = 10, noise = 0, r = 0, g = 0, b = 255, a = 75, speed = 0, receiver = 0, bool:reliable = true)
```

Comment:
I can't understand your preferences for default values, same below.

Reply:
@WPMGPRoSToTeMa I don't see what the problem is. I set the duration to 1 second in all natives, chose a blue color for most of them and kept the other parameters similar in al the natives (e.g. width is 10 everywhere). As you can see from the video, the beams have a perfect set of parameters for displaying the default sprite as most people would use it. I can't just put 0 in all of them, there's no point of having them optional if I do that.

Reply:
@OciXCrom I think better just add a note with an example of using it.

Reply:
@WPMGPRoSToTeMa No, one of the goals of these natives is to prevent the usage of all ~15 arguments. Most of the times you won't need all of them, so having default values that have the most common settings is a huge help. Take `set_hudmessage` for example - it has totally random default values.